scripts/validate_min_versions_in_sync.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+"""
+Check pandas required and optional dependencies are synced across:
+
+ci/deps/actions-.*-minimum_versions.yaml
+pandas/compat/_optional.py
+
+TODO: doc/source/getting_started/install.rst
+
+This is meant to be run as a pre-commit hook - to run it manually, you can do:
+
+    pre-commit run validate-min-versions-in-sync --all-files
+"""
+from __future__ import annotations
+
+import pathlib
+import sys
+
+DOC_PATH = pathlib.Path("doc/source/getting_started/install.rst").resolve()
+CI_PATH = next(
+    pathlib.Path("ci/deps").absolute().glob("actions-*-minimum_versions.yaml")
+)
+CODE_PATH = pathlib.Path("pandas/compat/_optional.py").resolve()
+# pandas package is not available
+# in pre-commit environment
+sys.path.append("pandas/compat")
+sys.path.append("pandas/util")
+import version
+
+sys.modules["pandas.util.version"] = version
+import _optional
```

Comment:
can you import these as fully qualified and/or make this whole section a bit more idiomatic (i know tricky, but give a try)

Reply:
Tried playing around with this but couldn't simplify it further than what @lithomas1 suggested.